### ISSUE TITLE
[REQUEST] Bad Proxy System

### ISSUE THREAD

**Issue by Unkn9wn1** [openbullet/openbullet#468]
Hi bro,
i'm using OB from first releases and before that many other programs from 2014 until now
and i tested most of programs like OB
i think proxy system in OB is against OB's Runner's system.
there is no problem for guys who use only one config on OB
but there is many many problems for me and guys like me
because we run several Runners in same time
and that all is because:
proxy settings and also proxies is different for each config

and also there is more problems because most of proxy options is in general settings
instead of config's proxy settings

- i have proposal for you to make it OK, without making old configs broken.

+ Add ability to add several proxy lists in OB [ Detailed before in #448 ]

Move this options : {
Allow Concurrent Use
Never ban Proxies
Shuffle Proxies on Start
Ban Loop Evasion
Don't Reuse Clearance Cookie
Reload proxies when all banned
} to Config's Proxy Options (this options are diff in each config)

Move this options : {
Shuffle Proxies on Start
} to "Proxies" tab in each proxy list's "Detailed View"

+ integrate "Proxy Source" to "Proxies" tab and add a update time to automatic refresh list from remote source or even from file and add option here to auto remove duplicates when got proxies from the source / sources

+ Remove Proxy BAD Condition, Instead (
- Add option to "Ban a proxy if the number of tested credentials/data are greater than {-}" (result not matter)
it's useful for sites that bans if an IP try to login for minutes and if you try again they will adds ban-time or even bans IP/Proxy for days

- Add "timeout before unban all banned proxies"
It's useful for targets that ban proxies for specified time, or even make proxies harder to ban
)
Bad condition is not useful when using good proxies and it's good to remove it because OB has ability to check proxies before use and this makes it bad for real

if you even don't want to removed BAD Condition, please add "timeout before unban" and "ban proxy if tested creds/data are greater than" options

it could be a good step to walk in version 2!

**Comment by openbullet** [comment]
I understand, this is indeed a good idea, but this is a lot of work and I barely have any time to code as of late so don't expect anything soon.

**Comment by Unkn9wn1** [comment]
Yes, i don't to expect to see big changes soon, but i just test OB this days and i write my ideas to make it greater with my great ideas. lol

i also edited this issue read it again please, thank you

**Comment by Cyber-Criminal** [comment]
a Blacklist is good too, add proxy to blacklist after Hit (Multiple Hits with same proxy may cause ban accounts), and blacklist proxy after many retries without any checked creds
